modules/glean/runtime/glean-plugin.client.ts

```diff
@@ -11,10 +11,13 @@ export default defineNuxtPlugin((nuxtApp) => {
     log.info('Glean: App mounted, start initing glean')
 
     const GLEAN_APP_ID = 'mozilla-social-web'
-    const devMode = useAppConfig().env === ('dev' || 'canary' || 'preview')
-    const uploadEnabled = devMode // this will eventually be a setting that the user can toggle
+    const env = useAppConfig().env
+    const devMode = env === ('dev' || 'canary' || 'preview')
+    const userSettings = useUserSettings()
+    const allowGlean = getPreferences(userSettings.value, 'allowGlean')
+    const uploadEnabled = devMode && allowGlean
```

Comment:
Should this be an OR instead of AND?

Reply:
Right now, no - keeping the `uploadEnabled` for dev only until we are ready to start collecting events in production